src/rgw/rgw_civetweb.cc

```diff
@@ -120,6 +120,10 @@ void RGWCivetWeb::init_env(CephContext *cct)
     env.set("REMOTE_USER", info->remote_user);
   }
 
+  if (info->remote_addr) {
```

Comment:
Would it be better to check the string length too ?

`strlen(info->remote_addr) != 0` 
or 
`if(info->remote_addr[0] != '\0')`